manifests/certonly.pp

```diff
@@ -44,6 +44,7 @@
   Boolean                 $suppress_cron_output = false,
   Optional[String[1]]     $cron_before_command  = undef,
   Optional[String[1]]     $cron_success_command = undef,
+  Optional[Array]         $cron_monthday        = ['*'],
```

Comment:
Since you provide a default, this can just be `Array`. Should it also be `Array[String[1]]`?

Reply:
Sure, we could omit the `Optional` and have just an `Array` instead.  If we use `Array[String[1]]` then one needs to specify the days in the `cron_monthday` parameter as strings like `['1', '15']`. `[1, 15]` will not work because puppet will complain with `expects a String value, got Integer`. What do you think is better? Having `Array` or `Array[String[1]]` ?

Reply:
I'm a fan of strict data types so perhaps `Array[Variant[Integer, String[1]]]`

Reply:
I couldn't agree more with you. Shall I submit a new commit to this PR including this change?